streams/src/main/java/org/apache/kafka/streams/kstream/internals/suppress/BufferConfigInternal.java

```diff
@@ -18,13 +18,29 @@
 
 import org.apache.kafka.streams.kstream.Suppressed;
 
+import java.util.HashMap;
+import java.util.Map;
+
 import static org.apache.kafka.streams.kstream.internals.suppress.BufferFullStrategy.SHUT_DOWN;
 
-abstract class BufferConfigInternal<BC extends Suppressed.BufferConfig<BC>> implements Suppressed.BufferConfig<BC> {
+public abstract class BufferConfigInternal<BC extends Suppressed.BufferConfig<BC>> implements Suppressed.BufferConfig<BC> {
```

Comment:
I browsed a bit, but couldn't figure out why this needs to be public now. Can you fill me in?

Reply:
At `KTableImpl.java` L410,411 ([here](https://github.com/apache/kafka/blob/f32369cd4a0052d427af20aedaf44c707625dc7f/streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java#L410)) BufferConfigInternal is used (although the type is not explicitly imported) which requires this class to be public

Reply:
Ah, there it is. Thanks for the clarification.